### PR TITLE
change to node CLI tool

### PR DESCRIPTION
**Pull request by obiwanzenobi**

### CODE REVIEW

File: convert.js [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+}
+
+function getFillFromNode(node) {
+    var fill = '';
+
+    if (node.attributes.fill != null) {
+        fill = node.attributes.fill;
+    }
+    else {
+        fill = node.attributes.fill;
```
Comment (flutter-clutter) [inline convert.js]: Something is broken here. It does the same in both conditions. Opacity is also broken.

**Review by flutter-clutter — COMMENTED** [review]
Hey there! Thanks a lot for your effort.

Some things don't work anymore like opacity and fill color. I am about to restructure the code based on your very great changes. I hope you don't mind. Think it will take me some days. :)

**Comment by obiwanzenobi** [comment]
Sure thing, if you could send me the svg files that you had problem with, i could make a fix for that.

File: convert.js (hunk 1 above)
Comment (obiwanzenobi) [reply convert.js]: I'll check that

**Comment by flutter-clutter** [comment]
@obiwanzenobi 

> Sure thing, if you could send me the svg files that you had problem with, i could make a fix for that.

Thank you of the offer :). In fact, I've already solved this problem. It's due to the fact that the lib you chose (`svgson`) at least in the version defined in the package.json returns a string when accessing the `style` attribute of a `node`. In the previous version of my path converter, it was an object, so the access via `path.node.attributes.style.fill` was not an issue. I solved it by creating a function that converts the style string to an object:

```
function styleStringToObject(styleString) {
  let regex = /([\w-]*)\s*:\s*([^;]*)/g;
  let match, properties={};
  while(match=regex.exec(styleString)) properties[match[1]] = match[2].trim();

  return properties;
}
```

Will commit it together with my other changes the next days.

**Comment by flutter-clutter** [comment]
@obiwanzenobi What do you think about the current version? Would you be willing to test it with a few SVGs?

**Comment by obiwanzenobi** [comment]
@flutter-clutter Sorry, been busy for a couple of days.
Looks much cleaner! I have some problems with stroke-only paths (i get 0 opacity paints) but I think this is something to work on after we merge that one.

**Comment by obiwanzenobi** [comment]
0 opacity problem is also happening when there is no fill and no stroke parameter defined like in the first w3c svg example:
`<svg height="210" width="400">
  <path d="M150 0 L75 200 L225 200 Z" />
</svg>`

https://www.w3schools.com/graphics/svg_path.asp

By default web browsers renders it as a black filling

**Pull request merged**